File: NQT/Arrays/BinaryRowMax1s.java
import java.io.*;

public class BinaryRowMax1s {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        System.out.println("Enter the number of rows:");
        int rows = Integer.parseInt(reader.readLine());
        System.out.println("Enter the number of columns:");
        int cols = Integer.parseInt(reader.readLine());
        int[][] matrix = new int[rows][cols];
        System.out.println("Enter the binary matrix (0s and 1s):");
        for (int i = 0; i < rows; i++) {
            String[] input = reader.readLine().split(" ");
            for (int j = 0; j < cols; j++) { 
                matrix[i][j] = Integer.parseInt(input[j]);
            }
        }
        reader.close();
        int maxOnesRowIndex = findRowWithMaxOnes(matrix, rows, cols);
        if (maxOnesRowIndex != -1) {
            System.out.println("Row with maximum number of 1s: " + maxOnesRowIndex);
        } else {
            System.out.println("No row contains 1s.");
        }
    }

    private static int findRowWithMaxOnes(int[][] matrix, int rows, int cols) {
        int maxOnesRowIndex = -1;
        int maxOnesCount = 0;
        for (int i = 0; i < rows; i++) {
            int onesCount = countOnesInRow(matrix[i], cols);
            if (onesCount > maxOnesCount) {
                maxOnesCount = onesCount;
                maxOnesRowIndex = i;
            }
        }
        return maxOnesRowIndex;
    }

    private static int countOnesInRow(int[] row, int cols) {
        int count = 0;
        for (int j = 0; j < cols; j++) {
            if (row[j] == 1) {
                count++;
            }
        }
        return count;
    }
}

File: NQT/Arrays/Python/binary_row_max_1s.py
# find the row with the maximum number of 1s in a binary matrix
def main():
    print("Enter the number of rows and columns in the binary matrix (space-separated):")
    rows, cols = map(int, input().split())
    
    print(f"Enter the elements of the binary matrix row by row (each row separated by a new line):")
    binary_matrix = []
    for _ in range(rows):
        row = list(map(int, input().split()))
        binary_matrix.append(row)
    
    max_row_index = find_row_with_max_1s(binary_matrix)
    
    if max_row_index != -1:
        print(f"The row with the maximum number of 1s is at index: {max_row_index}")
    else:
        print("No rows found with 1s.")

def find_row_with_max_1s(matrix):
    max_count = 0
    max_row_index = -1
    
    for i, row in enumerate(matrix):
        count_1s = sum(row)
        if count_1s > max_count:
            max_count = count_1s
            max_row_index = i
            
    return max_row_index


if __name__ == "__main__":
    main()